14.0 t2015 purchase vendor bill product breakdown create new module

geomer198 opened the pull request.

LoisRForgeFlow committed:
product_supplier_code_purchase

oca-transbot committed:
OCA Transbot updated translations from Transifex

nikul-serpentcs committed:
[MIG] product_supplier_code_purchase v10

AaronHForgeFlow committed:
[IMP]computing purchase supplier code

oca-travis committed:
[UPD] Update product_supplier_code_purchase.pot

mvrodriguez committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (2 of 2 strings)

Translation: purchase-workflow-10.0/purchase-workflow-10.0-product_supplier_code_purchase
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-10-0/purchase-workflow-10-0-product_supplier_code_purchase/es/

OCA-git-bot committed:
[ADD] icon.png

AdriaGForgeFlow committed:
[12.0][ADD] purchase_manual_delivery

[ADD] ability to change destination location in wizard

AaronHForgeFlow committed:
[MIG] product_supplier_code_purchase to v12

oca-travis committed:
[UPD] Update purchase_manual_delivery.pot

OCA-git-bot committed:
[UPD] README.rst

oca-travis committed:
[UPD] Update product_supplier_code_purchase.pot

OCA-git-bot committed:
[UPD] README.rst

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-12.0/purchase-workflow-12.0-product_supplier_code_purchase
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-product_supplier_code_purchase/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-12.0/purchase-workflow-12.0-product_supplier_code_purchase
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-product_supplier_code_purchase/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 66.7% (2 of 3 strings)

Translation: purchase-workflow-12.0/purchase-workflow-12.0-product_supplier_code_purchase
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-product_supplier_code_purchase/sl/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-12.0/purchase-workflow-12.0-product_supplier_code_purchase
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-product_supplier_code_purchase/

SandieFavre committed:
Translated using Weblate (French)

Currently translated at 100.0% (3 of 3 strings)

Translation: purchase-workflow-12.0/purchase-workflow-12.0-product_supplier_code_purchase
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-product_supplier_code_purchase/fr/

marcelsavegnago committed:
Translated using Weblate (Portuguese (Brazil))

Currently translated at 33.3% (1 of 3 strings)

Translation: purchase-workflow-12.0/purchase-workflow-12.0-product_supplier_code_purchase
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-product_supplier_code_purchase/pt_BR/

sergio-teruel committed:
[IMP] purchase_order_secondary_unit: Less code. Use product_secondary_unit mixin model from product-attribute repository.

sysadminmatmoz committed:
Added translation using Weblate (Slovenian)

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (56 of 56 strings)

Translation: purchase-workflow-12.0/purchase-workflow-12.0-purchase_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-purchase_manual_delivery/sl/

sysadminmatmoz committed:
Translated using Weblate (Slovenian)

Currently translated at 100.0% (3 of 3 strings)

Translation: purchase-workflow-12.0/purchase-workflow-12.0-product_supplier_code_purchase
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-product_supplier_code_purchase/sl/

Zar21 committed:
[IMP] product_supplier_code_purchase: black, isort, prettier

Zar21 committed:
[MIG] product_supplier_code_purchase: Migration to 14.0

kittiu committed:
[14.0][IMP] purchase_ivoice_plan, non-editable installment

newtratip committed:
[FIX] split function _prepare_advance_purchase_line for hook

ChrisOForgeFlow committed:
[IMP] purchase_manual_delivery: black, isort, prettier

ChrisOForgeFlow committed:
[14.0][MIG] purchase_manual_delivery

ChrisOForgeFlow committed:
[ADD] configuration to make pickings on purchase orders

ChrisOForgeFlow committed:
[FIX] permissions

ChrisOForgeFlow committed:
IMP: added test for fields on purchase order line

ChrisOForgeFlow committed:
[FIX] rounding with decimal precision

ChrisOForgeFlow committed:
[FIX] added field invisible to ensure wizard works

victoralmau committed:
[ADD] purchase_order_price_recalculation: New addon.
TT33369

simplify-m110 committed:
[14.0][FIX] purchase_request _compute_open_product_qty

aayartsev committed:
[FIX] fix product domain in PO

[14.0][FIX] purchase_order_vendor_product, fix pre-commit error

[UPD] README.rst

[ADD] Inherit _name_search method to modify domain

[FIX] Fixes with pre-commit

[12.0][ADD] purchase_manual_delivery

[ADD] ability to change destination location in wizard

[UPD] Update purchase_manual_delivery.pot

[UPD] README.rst

Added translation using Weblate (Slovenian)

Translated using Weblate (Slovenian)

Currently translated at 100.0% (56 of 56 strings)

Translation: purchase-workflow-12.0/purchase-workflow-12.0-purchase_manual_delivery
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-12-0/purchase-workflow-12-0-purchase_manual_delivery/sl/

[IMP] purchase_manual_delivery: black, isort, prettier

[14.0][MIG] purchase_manual_delivery

[ADD] configuration to make pickings on purchase orders

[FIX] permissions

IMP: added test for fields on purchase order line

[FIX] rounding with decimal precision

[FIX] added field invisible to ensure wizard works

[UPD] Update purchase_manual_delivery.pot

[UPD] addons table in README.md

[UPD] README.rst

[ADD] setup.py

[FIX] split function _prepare_advance_purchase_line for hook

purchase_deposit 14.0.1.1.1

[UPD] addons table in README.md

[ADD] add tests

[FIX] pre-commit fixes

[FIX] rename test function

oca-travis committed:
[UPD] Update purchase_manual_delivery.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
purchase_deposit 14.0.1.1.1

OCA-git-bot committed:
[UPD] addons table in README.md

oca-travis committed:
[UPD] Update purchase_invoice_plan.pot

OCA-git-bot committed:
purchase_invoice_plan 14.0.1.2.0

OCA-git-bot committed:
[UPD] addons table in README.md

oca-travis committed:
[UPD] Update purchase_order_vendor_product.pot

OCA-git-bot committed:
purchase_order_vendor_product 14.0.1.1.0

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-14.0/purchase-workflow-14.0-purchase_order_vendor_product
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-14-0/purchase-workflow-14-0-purchase_order_vendor_product/

MiquelRForgeFlow committed:
[FIX] purchase_triple_discount: decimal_precision is obsolete

OCA-git-bot committed:
purchase_triple_discount 14.0.1.0.1

OCA-git-bot committed:
[UPD] addons table in README.md

oca-travis committed:
[UPD] Update purchase_order_price_recalculation.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

geomer198 committed:
[14.0][ADD] purchase_vendor_bill_product_breakdown

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/res_partner.py:
```diff
@@ -0,0 +1,7 @@
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    bill_components = fields.Boolean(string="Add Invoice Brakeburn")
```
Vendor Bill Breakdown

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/product_supplierinfo_component.py:
```diff
@@ -0,0 +1,13 @@
+    )
+    partner_id = fields.Many2one("res.partner", related="supplierinfo_id.name")
+    component_id = fields.Many2one("product.product", string="Component", required=True)
+    qty = fields.Float(string="Qty per Unit", required=True)
```
product_uom_qty

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/product_supplierinfo_component.py:
```diff
@@ -0,0 +1,13 @@
+    partner_id = fields.Many2one("res.partner", related="supplierinfo_id.name")
+    component_id = fields.Many2one("product.product", string="Component", required=True)
+    qty = fields.Float(string="Qty per Unit", required=True)
+    uom_id = fields.Many2one("uom.uom", string="Unit od Measure", required=True)
```
product_uom_id

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order.py:
```diff
@@ -0,0 +1,26 @@
+    _inherit = "purchase.order"
+
+    bill_components = fields.Boolean(
+        string="Add Invoive Brakeburn",
```
Vendor Bill Breakdown

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order.py:
```diff
@@ -0,0 +1,26 @@
+        string="Add Invoive Brakeburn",
+    )
+
+    def set_partner_bill_components(self):
```
@onchange("partner_id") уже отменили?)

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order_line.py:
```diff
@@ -0,0 +1,81 @@
+        )
+        return suppliers[0] if len(suppliers) > 0 else False
+
+    def _get_id_and_create_component(self, component):
```
Название функции не понятно. Равно как и описание. А также непонятно, почему используется create внутри функции, которая вызывается в цикле.

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order_line.py:
```diff
@@ -0,0 +1,81 @@
+        supplierinfo = self.get_supplier()
+        if not supplierinfo:
+            return False
+        components = [
```
Зачем весь этот цирк с кучей create и затем ещё write сверху?
Если уж ты используешь create, почему сразу не добавить line_id при создание?
К тому же метод create поддерживает создание сразу нескольких записей за один вызов create([vals])

Ну и если уж так хочется сделать write, то почему не сделать  ({"component_ids": [(0, 0, {"key": value})]} итд

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order_line.py:
```diff
@@ -0,0 +1,81 @@
+    def _update_purchase_order_line_component(self):
+        """
+        Delete and create all purchase order line component
+        by supplier components
```
Updates purchase order line components based on supplierinfo

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order_line.py:
```diff
@@ -0,0 +1,81 @@
+            }
+        ).id
+
+    def _update_purchase_order_line_component(self):
```
_update_purchase_order_line_component**s**

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order_line_component.py:
```diff
@@ -0,0 +1,31 @@
+        "purchase.order.line", string="Purchase Order Line", reuired=True
+    )
+    component_id = fields.Many2one("product.product", string="Component", required=True)
+    qty = fields.Float(string="Qty/unit", required=True)
```
product_uom_qty

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order_line_component.py:
```diff
@@ -0,0 +1,31 @@
+    component_id = fields.Many2one("product.product", string="Component", required=True)
+    qty = fields.Float(string="Qty/unit", required=True)
+    total_qty = fields.Float(string="Total Qty", required=True)
+    uom_id = fields.Many2one("uom.uom", string="Unit od Measure", required=True)
```
product_uom_id

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order_line_component.py:
```diff
@@ -0,0 +1,31 @@
+
+    def _update_qty(self, qty_received):
+        """
+        Update Total Quantity by Quantity Received
```
Update Total Quantity based on Quantity Received

ivs-cetmix reviewed (changes requested):
См комментарии и активити к задаче

ma-landoo committed:
Added translation using Weblate (Dutch)

manengel committed:
Added translation using Weblate (German)

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order_line_component.py:
```diff
@@ -0,0 +1,39 @@
+    @api.model
+    def create(self, vals):
+        """Set state Done for component"""
+        vals["state"] = "done"
```
в чем смысл данного действия?

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/views/purchase_order_views.xml:
```diff
@@ -0,0 +1,83 @@
+                    <field
+                        name="component_ids"
+                        widget="one2many"
+                        context="{'default_state': 'draft'}"
```
У тебя и так это поле draft по умолчанию

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/product_supplierinfo.py:
```diff
@@ -0,0 +1,27 @@
+
+    bill_components = fields.Boolean(related="name.bill_components")
+    component_ids = fields.One2many("product.supplierinfo.component", "supplierinfo_id")
+    is_access = fields.Boolean(
```
Зачем это поле?

ivs-cetmix reviewed (changes requested):
Зачем ты изобретаешь велосипед?
Права доступа задаются через ir.model.access

ivs-cetmix commented on purchase_order_line_menu/__manifest__.py:
```diff
@@ -11,7 +11,7 @@
     "category": "Inventory/Purchase",
     "data": [
-        "views/purchase_order_line_views.xml",
+        "views/purchase_order_line_component_views.xml",
```
Почему изменения в этом модуле?

ivs-cetmix commented on purchase_product_usage/__manifest__.py:
```diff
@@ -13,7 +13,7 @@
         "security/purchase_product_usage.xml",
         "security/ir.model.access.csv",
-        "views/purchase_order_line_view.xml",
+        "views/purchase_order_line_component_views.xml",
```
Почему изменения в этом модуле?

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/product_supplierinfo.py:
```diff
@@ -0,0 +1,24 @@
+    bill_components = fields.Boolean(related="name.bill_components")
+    component_ids = fields.One2many("product.supplierinfo.component", "supplierinfo_id")
+
+    def action_test_data(self):
```
Странное название)
Видимо action_open_component_view?

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/product_supplierinfo.py:
```diff
@@ -0,0 +1,24 @@
+            "name": _("Product Components"),
+            "type": "ir.actions.act_window",
+            "view_mode": "form",
+            "res_model": self._name,
```
В принципе тут можно явно указать модель

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order.py:
```diff
@@ -0,0 +1,25 @@
+
+    @api.onchange("partner_id")
+    def set_partner_bill_components(self):
+        """Set bill components by partner field value 'bill components'"""
```
Set "Bill components" based on partner settings

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order.py:
```diff
@@ -0,0 +1,25 @@
+            rec.bill_components = rec.partner_id.bill_components
+
+    def _prepare_invoice(self):
+        """Prepare invoice with invoice lines by components"""
```
Prepare invoice values components based

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order_line.py:
```diff
@@ -0,0 +1,140 @@
+            "target": "new",
+        }
+
+    def _prepare_by_components_account_move_lines(self, move=False):
```
prepare_component_account_move_line

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order_line.py:
```diff
@@ -0,0 +1,140 @@
+        }
+
+    def _prepare_by_components_account_move_lines(self, move=False):
+        """Prepare by components account move lines"""
```
Prepare account move lines for components

ivs-cetmix commented on purchase_vendor_bill_product_breakdown/models/purchase_order_line.py:
```diff
@@ -0,0 +1,140 @@
+        return lines
+
+    def _prepare_account_move_line(self):
+        """Mark lines for skip"""
```
Mark lines to skip

ivs-cetmix reviewed (changes requested):
Нужно внести правки

kh-badep committed:
Added translation using Weblate (French)

oca-travis committed:
[UPD] Update product_supplier_code_purchase.pot

OCA-git-bot committed:
[UPD] addons table in README.md

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[ADD] setup.py

OCA-git-bot committed:
[UPD] addons table in README.md

HviorForgeFlow committed:
[FIX] po_secondary_unit: Drop 0 qty test check on pol with secondary UoM

oca-travis committed:
[UPD] Update purchase_order_secondary_unit.pot

OCA-git-bot committed:
purchase_order_secondary_unit 14.0.2.0.1

OCA-git-bot committed:
[UPD] addons table in README.md

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: purchase-workflow-14.0/purchase-workflow-14.0-purchase_order_secondary_unit
Translate-URL: https://translation.odoo-community.org/projects/purchase-workflow-14-0/purchase-workflow-14-0-purchase_order_secondary_unit/

OCA-git-bot committed:
purchase_request 14.0.1.3.5

OCA-git-bot committed:
[UPD] addons table in README.md

kittiu committed:
[14.0][FIX] purchase_invoice_plan, clean recompute tax dr/cr

OCA-git-bot committed:
purchase_invoice_plan 14.0.1.2.1

OCA-git-bot committed:
[UPD] addons table in README.md

ivs-cetmix committed:
Merge branch 'OCA:14.0' into 14.0-t2015-purchase_vendor_bill_product_breakdown-create-new-module

ivs-cetmix committed:
Merge branch '14.0-t2015-purchase_vendor_bill_product_breakdown-create-new-module' of https://github.com/cetmix/purchase-workflow into 14.0-t2015-purchase_vendor_bill_product_breakdown-create-new-module
